### DISCUSSION TITLE
Curious Search Results

### DISCUSSION THREAD

**Discussion by bfelz-da17** [microsoft-search/pnp-modern-search#3473] [Q&A]
When I set this up and secure it the site by guid or SPSite|Sitename I still get results from other sites.  In addition to that when I test this using basic end user credentials I get search results from other users onedrive.  Example...

If user jsmith enters a search term of jdoe it will return results from jdoe ondrive.

**Comment by kasperbolarsen** [comment]
Hi,
We really need some context here.
- Are you using SharePoint or Microsoft search as the source?
- Which query are you using?
- Have you checked if the results are identical to what you get from the API ( see https://microsoft-search.github.io/pnp-modern-search/#troubleshooting and perhaps use one of the tools mentioned)
-

**Comment by gretchunkim** [comment]
What you are seeing is by Microsoft Search engine design. It is suppose to return search results where the current user have been granted access to. This  can includes  Outlook emails, MS teams, OneDrive, External sources (if it's being ingested into MS Search),  and more (SharePoint). 

If you want to limit the search scope, then you have to specifically select site level, library level and other conditions  to do so.
